crates/containerd-shim-wasm/Cargo.toml

```diff
@@ -51,17 +50,17 @@ tracing-subscriber = { version = "0.3", default-features = false, features = [
 ], optional = true }
 
 # opentelemetry
-opentelemetry = { version = "0.23", features = ["trace"], optional = true }
-opentelemetry-otlp = { version = "0.16.0", features = [
+opentelemetry = { version = "0.26", features = ["trace"], optional = true }
+opentelemetry-otlp = { version = "0.26", features = [
     "grpc-tonic",
     "http-proto",
     "reqwest-client",
     "trace",
 ], optional = true }
-opentelemetry_sdk = { version = "0.23", features = [
+opentelemetry_sdk = { version = "0.26", features = [
```

Comment:
since we are no longer importing trace in this PR, can we disable default features for opentelemetry_sdk?

Reply:
This is the smallest subset of features that compiles
```toml
# tracing
# note: it's important to keep the version of tracing in sync with tracing-subscriber
tracing = { workspace = true, optional = true }
# does not include `tracing-log` feature due to https://github.com/spinkube/containerd-shim-spin/issues/61
tracing-subscriber = { version = "0.3", default-features = false, features = [
    "env-filter",
], optional = true }

# opentelemetry
opentelemetry = { version = "0.26", default-features = false, optional = true }
opentelemetry-otlp = { version = "0.26", default-features = false, features = [
    "grpc-tonic",
    "http-proto",
], optional = true }
opentelemetry_sdk = { version = "0.26", default-features = false, features = [
    "rt-tokio",
], optional = true }
tracing-opentelemetry = { version = "0.27", default-features = false, optional = true }
```

I don't know if some of the removed features have runtime side effects that don't affect compilation.

Reply:
we need to have an integration test for the tracing / otel work, but being able to compile gave me confident to approve this PR at least.

Reply:
Marking as draft until we resolve what features we need.

Reply:
Ok, based on:
* https://docs.rs/opentelemetry-otlp/0.26.0/opentelemetry_otlp/#feature-flags
* https://docs.rs/tracing-subscriber/latest/tracing_subscriber/index.html#feature-flags
* https://docs.rs/opentelemetry/latest/opentelemetry/#crate-feature-flags

I think the reduced set of flags above is fine, but I would include the `smallvec` (currently enabled) and `parking_lot` features to `tracing-subscriber` as they enable performance optimizations.
IIUC, the rest of the features make the crate export extra functionalities that we don't use.